### PR TITLE
Bernede1/astyle 3.1

### PR DESCRIPTION
**Pull request by adrienbernede**
Bump astyle and apply new style to check the differences.

 | PR | Author | Editor | Reviewers | Assignment | Approval | Merge| 
 | --- | --- | --- | --- | --- | --- | --- | 
| [#2506](https://github.com/mfem/mfem/pull/2506) | @adrienbernede | @tzanio | @tzanio + @v-dobrev | 09/05/21 | 09/13/21 | 09/13/21 | |

### CODE REVIEW

File: fem/intrules.cpp [hunk 1]
```diff
@@ -955,7 +955,7 @@ const IntegrationRule &IntegrationRules::Get(int GeomType, int Order)
             IntegrationRule *ir = GenerateIntegrationRule(GeomType, Order);
             int RealOrder = Order;
             while (RealOrder+1 < ir_array->Size() &&
-            /*  */ (*ir_array)[RealOrder+1] == ir)
+                   /*  */ (*ir_array)[RealOrder+1] == ir)
```
Comment (adrienbernede) [inline fem/intrules.cpp:958]: Looks like Astyle is now applying what the comment was enforcing.

File: fem/tmop_amr.cpp [hunk 2]
```diff
@@ -234,7 +234,7 @@ void TMOPRefinerEstimator::SetTriIntRules()
 
    // Reftype = 0 // original element
    const int Nvert = 3, NEsplit = 1;
-   Mesh meshsplit(2, Nvert, NEsplit, 0 ,2);
+   Mesh meshsplit(2, Nvert, NEsplit, 0,2);
```
Comment (adrienbernede) [inline fem/tmop_amr.cpp:237]: Ideally it would have put a space between ',2'.

File: mesh/mesh.cpp [hunk 3]
```diff
@@ -2850,14 +2850,14 @@ void Mesh::Make3D(int nx, int ny, int nz, Element::Type type,
          y = sfc[3*k + 1];
          z = sfc[3*k + 2];
 
-         ind[0] = VTX(x  , y  , z  );
-         ind[1] = VTX(x+1, y  , z  );
+         ind[0] = VTX(x, y, z  );
+         ind[1] = VTX(x+1, y, z  );
```
Comment (adrienbernede) [inline mesh/mesh.cpp:2854]: Some will probably be annoyed by this...

File: mesh/ncmesh.cpp [hunk 4]
```diff
@@ -612,7 +612,7 @@ int NCMesh::NewSegment(int n0, int n1, int attr, int vattr1, int vattr2)
    // get (degenerate) faces and assign face attributes
    int v0 = el.node[0], v1 = el.node[1];
    faces.Get(v0, v0, v0, v0)->attribute = vattr1;
-   faces.Get(v1, v1, v1 ,v1)->attribute = vattr2;
+   faces.Get(v1, v1, v1,v1)->attribute = vattr2;
```
Comment (adrienbernede) [inline mesh/ncmesh.cpp:615]:
```suggestion
   faces.Get(v1, v1, v1, v1)->attribute = vattr2;
```

File: fem/tmop_amr.cpp (hunk 2 above)
Comment (adrienbernede) [reply fem/tmop_amr.cpp:237]:
```suggestion
   Mesh meshsplit(2, Nvert, NEsplit, 0, 2);
```

File: fem/intrules.cpp (hunk 1 above)
Comment (adrienbernede) [inline fem/intrules.cpp:958]:
```suggestion
                   (*ir_array)[RealOrder+1] == ir)
```

File: fem/ceed/solvers-atpmg.cpp [hunk 5]
```diff
@@ -235,7 +235,7 @@ int CeedATPMGElemRestriction(int order,
                   // in the i, j, or k direction?
                   int in_edof = i*P1d*P1d + j*P1d + k;
                   int in_ldof = in_elem_dof[in_edof*in_layout[0] +
-                                            e*in_layout[2]] + rounding_guard;
+                                                                 e*in_layout[2]] + rounding_guard;
```
Comment (adrienbernede) [inline fem/ceed/solvers-atpmg.cpp:238]: Maybe the worst part of this reformatting.

**Comment by tzanio** [comment]
This PR is now [under review](https://github.com/mfem/mfem/blob/master/CONTRIBUTING.md#pull-requests) (see the table in the PR description). To help with the review process, please **do not force push** to the branch.

**Comment by tzanio** [comment]
@mfem/developers -- Heads up, this is another PR that will affect everyone. 

We need to upgrade the version of AStyle we are using due to GH Actions. The changes are not big, but please take a look and speak up if you have any concerns.

**Comment by tzanio** [comment]
Please mention in `CHANGELOG`

**Review by tzanio — APPROVED** [review]
Thanks for adding this @adrienbernede!

The changes are pretty minimal -- we should go for it 👍

**Review by vladotomov — APPROVED** [review]

**Review by camierjs — APPROVED** [review]

**Review by vladotomov — APPROVED** [review]

**Review by acfisher — APPROVED** [review]

**Review by vladotomov — APPROVED** [review]

**Comment by tzanio** [comment]
On 09/09/21 we decided to **make the switch on Monday, 09/13/21**

**Comment by adrienbernede** [comment]
So many approvals, what a popular PR :D

**Comment by tzanio** [comment]
Merged in `next` for testing...

**Pull request merged**